rust/lance-linalg/src/distance/norm_l2.rs

```diff
@@ -30,21 +30,40 @@ pub trait Normalize<T: Float> {
     fn norm_l2(&self) -> Self::Output;
 }
 
+// `avx512fp16` is not supported in rustc yet. Once it is supported, we can
+// move it to target_feture.
+#[cfg(any(
+    all(target_os = "macos", target_feature = "neon"),
+    feature = "avx512fp16"
+))]
+mod kernel {
+    use super::*;
+
+    extern "C" {
+        pub fn norm_l2_f16(ptr: *const f16, len: u64) -> f16;
+    }
+}
+
 impl Normalize<f16> for &[f16] {
     type Output = f16;
 
-    #[inline]
+    // #[inline]
```

Comment:
Should we remove this? Or keep it?

Reply:
Removed, did not figure out how to add it appropriately.